### PR TITLE
start using tox

### PR DESCRIPTION
**Pull request by rcerven**

### CODE REVIEW

**Comment by twaugh** [comment]
Can we add coveralls.io a badge to README.md?

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
+      env: TOXENV=py26
+    - python: "2.7"
+      env: TOXENV=py27
+    - python: "3.4"
+      env: TOXENV=py34
+    - python: "3.5"
+      env: TOXENV=py35
 # command to install dependencies
 install:
   - "pip install -r requirements.txt"
```
Comment (vrutkovs) [inline .travis.yml]: We can remove those lines since Tox now installs the dependencies

Comment (twaugh) [inline .travis.yml]: In theory setup.py would pull in the requirements from requirements.txt, and tox uses setup.py to install the package. This particular setup.py doesn't do that... in fact requirements.txt is empty.

Wonder if we should remove `- "pip install -r requirements.txt"` from this file and remove the requirements.txt file altogether, to make that clearer.

Comment (rcerven) [inline .travis.yml]: well it is empty for now, that might change, and this way it will work in future as well

Comment (vrutkovs) [inline .travis.yml]: >In theory setup.py would pull in the requirements from requirements.txt

This is probably the best option (see https://github.com/vrutkovs/github-review-dashboard/blob/master/setup.py#L15), but I'm not sure if py26 supports that

Comment (rcerven) [inline .travis.yml]: I am using same thing as is in osbs-client and atomic-reactor

File: setup.py [hunk 2]
```diff
@@ -8,9 +8,24 @@
 """
 
 import re
+import sys
 
 from setuptools import setup, find_packages
 
+def _get_requirements(path):
```
Comment (vrutkovs) [inline setup.py]: Manual parsing of `requirement.txt` is not a very good idea - it won't work in many cases, e.g.:
```
docker-py < 2.0  #New docker-py has a totally different API
```

File: setup.py [hunk 3]
```diff
@@ -20,5 +35,8 @@
     url='https://github.com/DBuildService/dockerfile-parse',
     license="BSD",
     packages=find_packages(exclude=["tests"]),
+    install_requires=_install_requirements(),
+    setup_requires=['pytest-runner'],
```
Comment (lcarva) [inline setup.py]: Keep in mind that this will fail RPM builds, this line must be nuked in spec file. Not sure about `tests_require`.

Comment (rcerven) [inline setup.py]: yeah I still need to change spec file

File: setup.py (hunk 2 above)
Comment (rcerven) [inline setup.py]: I tried to use same as in other components, so we can fine tune it later with something better, but as you said your idea doesn't work for 26

File: setup.py [hunk 4]
```diff
@@ -8,9 +8,24 @@
...
+        with open(path) as f:
+            packages = f.read().splitlines()
+    except (IOError, OSError) as ex:
+        raise RuntimeError("Can't open file with requirements: %s", repr(ex))
+    return [p.strip() for p in packages if not re.match(r"^\s*#", p)]
+
+def _install_requirements():
+    requirements = _get_requirements('requirements.txt')
+    if sys.version_info[0] >= 3:
+        requirements += _get_requirements('requirements-py3.txt')
```
Comment (lcarva) [inline setup.py]: Looks like `requirements-py3.txt` is an empty file. Place holder for the future?

Comment (vrutkovs) [inline setup.py]: I think its a good idea to include this so we could make `setup.py` similar across projects

Comment (rcerven) [inline setup.py]: yes, I've created it just to make it same in everywhere, and yeah for the future use

File: setup.py (hunk 3 above)
Comment (twaugh) [inline setup.py]: This change is still needed.

File: setup.py (hunk 2 above)
Comment (vrutkovs) [inline setup.py]: PipSession approach does work for py26

File: setup.py (hunk 3 above)
Comment (vrutkovs) [inline setup.py]: Can we move this to `tests_requires` so we wouldn't have to modify the spec file?

File: python-dockerfile-parse.spec [hunk 5]
```diff
@@ -74,6 +74,8 @@ Python 3 library for Dockerfile manipulation
 %prep
 %setup -n %{project}-%{commit}
 
+# TODO: Fix this properly upstream
+sed -i '/setup_requires/d' setup.py
```
Comment (twaugh) [inline python-dockerfile-parse.spec]: Not required now?

Comment (rcerven) [inline python-dockerfile-parse.spec]: yop forgot about that when I was reverting it back

**Pull request merged**